lib/rubocop/cop/style/exponential_notation.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+# TODO: when finished, run `rake generate_cops_documentation` to update the docs
+module RuboCop
+  module Cop
+    module Style
+      # This cop enforces consistency when using exponential notation
+      # for numbers in the code (eg 1.2e4). Multiple modes can be selected
```

Comment:
Multiple modes -> Different style are supported. 

That makes it consistent with RuboCop's terminology. I'd also add the explanations of the supported style before the examples - again, for consistency with other cops.